.github/workflows/ci.yml

```diff
@@ -47,51 +45,43 @@ jobs:
   setup:
     uses: ./.github/workflows/setup.yml
 
-  build_linux_packages:
-    name: Build Linux Packages
+  linux_build_and_test:
+    name: Build And Test Linux Packages
```

Comment:
These names are getting too long to be useful and they repeat some information, like `Build And Test Linux Packages (linux-mi300-1gpu-ossci-rocm, gfx94X-dcgpu, gfx942) / Build Linux Packages / Build Linux Packages`:

https://github.com/ROCm/TheRock/actions/runs/14843042280?pr=538
![image](https://github.com/user-attachments/assets/59dd611c-44ff-445f-b0d1-09bca1611dfb)

Here this could just say "Linux" instead of "Build And Test Linux Packages"

Information in rough order of importance:

* Operating system
* gfx target
* Friendly machine/runner name (e.g. mi300, not "linux-mi300-1gpu-ossci-rocm")
* Any other details (release vs debug build, sanitizers enabled, shard # for when splitting tests across multiple runners, etc.)